### PR TITLE
chore: improve code :shipit: 

### PR DESCRIPTION
**Pull request by MiracleHorizon**

### CODE REVIEW

**Comment by MiracleHorizon** [comment]
Жду код ревью

**Comment by MiracleHorizon** [comment]
Андрей

File: src/main.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
+
 import App from "./App";
 import "./index.css";
 
-ReactDOM.createRoot(document.getElementById("root")!).render(
+const rootEl = document.getElementById("root")!;
+ReactDOM.createRoot(rootEl).render(
```
Comment (LWJerri) [inline src/main.tsx]:
```suggestion
const rootElement = document.getElementById("root")!;

ReactDOM.createRoot(rootElement).render(
```

File: src/components/ui/tabs.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
-import { cn } from "@/lib/utils";
+import { forwardRef, type ElementRef, type ComponentPropsWithoutRef } from "react";
 import * as TabsPrimitive from "@radix-ui/react-tabs";
-import * as React from "react";
 
+import { cn } from "@/lib/utils";
+
+// FIXME: Unused?
```
Comment (LWJerri) [inline src/components/ui/tabs.tsx:6]: Components inside `ui` folder automatically generated from [shadcn/ui](https://ui.shadcn.com), so this comment here is unrelated, due I do not maintain this part of code.

File: src/components/custom/findid.tsx [hunk 3]
```diff
@@ -18,43 +28,53 @@ export function FindIDTab({ bg }: { bg: Function }) {
...
-          <Label htmlFor="source">Content that containts IDs*</Label>
+          <Label htmlFor={sourceId}>Content that contains IDs*</Label>
 
-          <Textarea id="source" value={sourceValue} onChange={(e) => setSourceValue(e.target.value)} />
+          <Textarea id={sourceId} value={sourceValue} onChange={handleChangeSourceValue} />
 
-          <p className="text-xs text-gray-500">*For example, it could be one or more codes or a message from Chime.</p>
+          <p className="text-xs text-gray-500">
+            *For example, it could be one or more codes or a message from Chime.
+          </p>
```
Comment (LWJerri) [inline src/components/custom/findid.tsx:66]: Why did you do that here?

File: src/components/custom/findid.tsx [hunk 4]
```diff
@@ -18,43 +28,53 @@ export function FindIDTab({ bg }: { bg: Function }) {
...
+  const handleEnterKey = (ev: KeyboardEvent<HTMLInputElement>) => {
+    if (ev.key !== "Enter") return;
 
     handleScanButton();
   };
 
+  const handleChangeSourceValue = (ev: ChangeEvent<HTMLTextAreaElement>) => setSourceValue(ev.target.value);
+  const handleChangeScannedValue = (ev: ChangeEvent<HTMLInputElement>) => {
+    setScannedValue(ev.target.value.replace(/ /g, ""));
+  };
```
Comment (LWJerri) [inline src/components/custom/findid.tsx]:
```suggestion
  const handleChangeSourceValue = (ev: ChangeEvent<HTMLTextAreaElement>) => setSourceValue(ev.target.value);

  const handleChangeScannedValue = (ev: ChangeEvent<HTMLInputElement>) => {
    setScannedValue(ev.target.value.replace(/ /g, ""));
  };
```

File: src/components/custom/findid.tsx [hunk 5]
```diff
@@ -18,43 +28,53 @@ export function FindIDTab({ bg }: { bg: Function }) {
     if (sourceValueIDs) {
       bg("bg-green-200");
 
-      new Audio("song.wav").play();
+      const audio = new Audio();
```
Comment (LWJerri) [inline src/components/custom/findid.tsx]:
```suggestion
      const audio = new Audio("song.wav");
```

File: src/components/custom/extractid.tsx [hunk 6]
```diff
@@ -1,68 +1,83 @@
...
 
-  const extractValues = (value: string) => {
+  const handleChangeSourceValue = (ev: ChangeEvent<HTMLTextAreaElement>) => setSourceValue(ev.target.value);
+
+  const handleChangeExtractValues = (ev: MouseEvent<HTMLButtonElement>) => {
+    const value = ev.currentTarget.value;
     setExtractor(value);
 
     const getDataFromSource = sourceValue.match(AVAILABLE_CHOICES[value].regexp);
+    if (!getDataFromSource) {
```
Comment (LWJerri) [inline src/components/custom/extractid.tsx:41]:
```suggestion

    if (!getDataFromSource) {
```

**Review by LWJerri — CHANGES_REQUESTED** [review]
Thanks for reviewing my code and making changes to improve the code! Please revise my notes regarding your PR.

**Review by LWJerri — APPROVED** [review]
